arangod/RocksDBEngine/RocksDBSettingsManager.cpp

```diff
@@ -114,184 +108,185 @@ void RocksDBSettingsManager::retrieveInitialValues() {
   _engine.releaseTick(_initialReleasedTick);
 }
 
-bool RocksDBSettingsManager::lockForSync(bool force) {
-  if (force) {
-    while (true) {
-      bool expected = false;
-      bool res = _syncing.compare_exchange_strong(
-          expected, true, std::memory_order_acquire, std::memory_order_relaxed);
-      if (res) {
-        break;
-      }
-      std::this_thread::sleep_for(std::chrono::milliseconds(10));
-    }
-  } else {
-    bool expected = false;
-
-    if (!_syncing.compare_exchange_strong(expected, true,
-                                          std::memory_order_acquire,
-                                          std::memory_order_relaxed)) {
-      return false;
-    }
+// Thread-Safe force sync.
+ResultT<bool> RocksDBSettingsManager::sync(bool force) {
+  TRI_IF_FAILURE("RocksDBSettingsManagerSync") {
+    return ResultT<bool>::success(false);
   }
 
-  return true;
-}
-
-/// Thread-Safe force sync
-Result RocksDBSettingsManager::sync(bool force) {
-  TRI_IF_FAILURE("RocksDBSettingsManagerSync") { return Result(); }
-  if (!_db) {
-    return Result();
+  std::unique_lock lock{_syncingMutex, std::defer_lock};
+  if (!_db || (!lock.try_lock() && !force)) {
+    return ResultT<bool>::success(false);
   }
-
-  if (!lockForSync(force)) {
-    return Result();
+  if (!lock) {
+    lock.lock();
   }
```

Comment:
NIT
```C++
if (!_db) {
  return ResultT<bool>::success(false);
}
std::unique_lock lock{_syncingMutex, std::defer_lock};
if (force) {
  lock.lock();
} else if (!lock.try_lock()) {
  return ResultT<bool>::success(false);  
}
```

Sorry for my old suggestion, but I think this more clear and better (lock instead of try_lock and then lock)
Also more clear that we check _db before create unique_lock